datanode/entities/volume_discount.go

```diff
@@ -0,0 +1,57 @@
+package entities
+
+import (
+	"fmt"
+	"time"
+
+	"code.vegaprotocol.io/vega/libs/num"
+	eventspb "code.vegaprotocol.io/vega/protos/vega/events/v1"
+)
+
+type (
+	VolumeDiscountStatsUpdated struct {
+		AtEpoch                  uint64
+		PartyVolumeDiscountStats []*PartyVolumeDiscountStats
```

Comment:
If you don't need to, I think you can just use the proto message here instead of having to map it out.

Reply:
happy to use it !